R/query.R

```diff
@@ -139,22 +143,25 @@ orderly_latest <- function(name = NULL, root = NULL, locate = TRUE,
                            draft = FALSE, must_work = TRUE) {
   config <- orderly_config(root, locate)
 
-  if (is.null(name)) {
-    d <- orderly_list2(draft, config, FALSE)
-    ids <- d$id
-    path <-
-      file.path((if (draft) path_draft else path_archive)(config$root), d$name)
-  } else {
-    path <-
-      file.path((if (draft) path_draft else path_archive)(config$root), name)
-    ids <- orderly_list_dir(path, check_run_rds = draft)
+  draft <- query_check_draft(draft)
+  path_funcs <- switch(draft,
+                       always = c(list_draft),
+                       never = c(list_archive),
+                       newer = c(list_draft, list_archive))
+  what <- switch(draft,
+                 always = "draft",
+                 never = "archive",
+                 newer = "draft or archive")
+
+  ids <- c()
```

Comment:
`c()` is `NULL` but I think here you mean `character(0)`

Reply:
alternatively can you do:

```
ids <- unlist(lapply(path_funcs, function(f) f(name, config)))
```

Reply:
> `c()` is `NULL` but I think here you mean `character(0)`

Oops yep, thanks - I think I must have done this many times before